pkgs/development/compilers/cudatoolkit/common.nix

```diff
@@ -199,6 +205,12 @@ stdenv.mkDerivation rec {
         patchelf --set-rpath "$rpath2" --force-rpath $i
       fi
     done < <(find $out $lib $doc -type f -print0)
+    ${lib.optionalString (lib.versionAtLeast version "11") ''
+      while IFS= read -r -d $'\0' i; do
```

Comment:
is there a reason you can't just use

```
for file in $out/target-linux-x64/*.so; do
  # stuff
done
```
?

I have no idea why the existing code uses a while-read-find mess.

Reply:
Good call, that seems much cleaner. Replaced with a for loop in 006d20e. The use of while-read-find in existing code might me more defensible since it's traversing recursively (but could probably still be cleaned up a bit)